src/cli/project/export/pypi_requirements.rs

```diff
@@ -0,0 +1,323 @@
+            entry.push_str("-e ");
+        }
+        entry.push_str(&self.source);
+
+        if let Some(hash) = &self.hash_flag {
+            entry.push_str(&format!(" {}", hash));
+        }
+
+        entry
+    }
+}
+
+fn get_pypi_hash_str(package_data: &PypiPackageData) -> Option<String> {
+    if let Some(hashes) = &package_data.hash {
+        let h = match hashes {
+            PackageHashes::Sha256(h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5Sha256(_, h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5(h) => format!("--hash=md5:{:x}", h).to_string(),
+        };
+        Some(h)
+    } else {
+        None
+    }
+}
+
+fn render_pypi_requirements(
+    target: impl AsRef<Path>,
+    packages: &[PypiPackageReqData],
+) -> miette::Result<()> {
+    if packages.is_empty() {
+        return Ok(());
+    }
+
+    let target = target.as_ref();
+
+    let reqs = packages
+        .iter()
+        .map(|p| p.to_req_entry())
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    fs::write(target, reqs)
+        .into_diagnostic()
+        .with_context(|| format!("failed to write requirements file: {}", target.display()))?;
+
+    Ok(())
+}
+
+fn render_env_platform(
+    output_dir: &Path,
+    env_name: &str,
+    env: &Environment,
+    platform: &Platform,
+    split_nohash: bool,
+) -> miette::Result<()> {
+    let packages = env.packages(*platform).ok_or(miette::miette!(
+        "platform '{platform}' not found for env {}",
+        env_name,
+    ))?;
+
+    let mut pypi_packages: Vec<PypiPackageReqData> = Vec::new();
+
+    for package in packages {
+        match package {
+            Package::Pypi(p) => pypi_packages.push(PypiPackageReqData::from_pypi_package(&p)),
+            Package::Conda(cp) => {
+                tracing::warn!(
+                    "ignoring Conda package {} since Conda packages are not supported in requirements.txt",
+                    cp.package_record().name.as_normalized()
+                );
+            }
+        }
+    }
+
+    let (base, nohash) = if split_nohash {
+        // Split package list based on presence of hash since pip currently treats requirements files
+        // containing any hashes as if `--require-hashes` has been supplied. The only known workaround
+        // is to split the dependencies, which are typically from vcs sources into a separate
+        // requirements.txt and to install it separately.
+        pypi_packages
+            .into_iter()
+            .partition(|p| p.editable || p.hash_flag.is_some())
+    } else {
+        (pypi_packages, vec![])
+    };
+
+    tracing::info!("Creating requirements file for env: {env_name} platform: {platform}");
+    let target = output_dir
+        .join(format!("{}_{}_requirements.txt", env_name, platform))
+        .into_os_string();
+
+    render_pypi_requirements(target, &base)?;
+
+    if !nohash.is_empty() {
+        tracing::info!(
```

Comment:
Info is only shown on `--verbose` is that what you want? If it should show on normal level you can either print it with `eprintln` or with `tracing::warn!`